quattor/types/aquilon.pan

```diff
@@ -14,8 +14,24 @@ type structure_fsdata = {
     "block_device_path" : string # No validation on this!
 };
 
-type structure_resources = extensible {
-    "filesystem"    ? structure_fsdata[]
+type structure_virtual_machine = extensible {
+    "name" : string
+    "hardware" : structure_hardware
+};
+
+type structure_resources = {
```

Comment:
any chance for some annotation? in particaulr, waht is the `virtual_machine` used for?

Reply:
If aquilon knows that a VM is on a particular hypervisor, that VM's hardware tree is copied into the hypervisor's profile under `/system/resources`

For example:
```
# ccm /system/resources/virtual_machine/1
+-/system/resources/virtual_machine/1
  +-hardware
    +-cards
      +-nic
        +-eth0
          $ boot : true
          $ hwaddr : 'aa:bb:cc:dd:ee:ff'
          $ media : 'Ethernet'
          $ name : 'Generic Network Interface'
          $ pxe : true
    +-cpu
      +-0
        $ arch : 'x86_64'
        $ cores : 1
        $ manufacturer : 'Intel'
        $ model : 'QEMU Virtual CPU version (cpu64-rhel6)'
        $ speed : 2000
      +-1
        $ arch : 'x86_64'
        $ cores : 1
        $ manufacturer : 'Intel'
        $ model : 'QEMU Virtual CPU version (cpu64-rhel6)'
        $ speed : 2000
    +-harddisks
      +-sda
        $ boot : true
        $ capacity : 10240
        $ interface : 'sata'
        $ model : 'Generic SATA disk'
        $ part_prefix : ''
    $ location : 'r89.harwell.europe'
    $ model : 'vm-openstack'
    $ nodename : 'vm-openstack-example-103'
    +-rack
      $ column : '420'
      $ name : 'r89rack420'
      $ room : 'example'
      $ row : '420'
    +-ram
      +-0
        $ size : 8192
    $ serialnumber : 'aaaabbbb-cccc-dddd-eeee-ffff00001111'
    +-sysloc
      $ building : 'r89'
      $ campus : 'harwell'
      $ city : 'harwell'
      $ continent : 'europe'
    $ type : 'singledisk'
  $ name : 'vm-openstack-example-103'
```